Training24/Test.c

```diff
@@ -90,97 +96,116 @@ void PrintTableHeader (enum HeaderType type) {
-      if (number < 0) IntegerTableRow (number, "Invalid(Negative number)", "Not a Palindrome", expectedOutputs[i]);
+      char* endPtr;
+      number = strtol (testCases[i], &endPtr, 10);
+      if (*endPtr != '\0') {
+         strcpy (outputStr, "Invalid(Non Integer)");
+         const char* actualOutput = "Not a Palindrome";
+         PrintIntegerTableRow (0, outputStr, expectedOutputs[i], actualOutput, testCases[i]);
+      }
       else {
-         overflow = ReverseAndCheckOverflow (number, &reversedNumber);
-         if (overflow) {
-            strcpy (outputStr, "Integer Overflow");
-            IntegerTableRow (number, outputStr, "Failed to process", expectedOutputs[i]);
+         if (number < 0) {
+            strcpy (outputStr, "Invalid(Negative number)");
+            const char* actualOutput = "Not a Palindrome";
+            PrintIntegerTableRow (number, outputStr, expectedOutputs[i], actualOutput, testCases[i]);
          }
          else {
-            sprintf (outputStr, "%d", reversedNumber);
-            const char* actualOutput = IsNumPalindrome (number) ? "Palindrome" : "Not a Palindrome";
-            IntegerTableRow (number, outputStr, actualOutput, expectedOutputs[i]);
+            overflow = ReverseAndCheckOverflow (number, &number);
+            if (!overflow) {
+               strcpy (outputStr, "Integer Overflow");
+               const char* actualOutput = "Failed to process";
+               PrintIntegerTableRow (number, outputStr, expectedOutputs[i], actualOutput, testCases[i]);
+            }
+            else {
+               sprintf (outputStr, "%d", number);
+               const char* actualOutput = IsIntPalindrome (number) ? "Palindrome" : "Not a Palindrome";
+               PrintIntegerTableRow (number, outputStr, expectedOutputs[i], actualOutput, testCases[i]);
+            }
          }
       }
    }
 }
 
-void TestUserPhrase () {
+void GetUserPhrase () {
    char userPhrase[500];
    while (1) {
-      printf ("\nEnter a phrase to test (max 499 characters, or type 'r' to return menu): ");
+      printf ("\nEnter a phrase to test (max 499 characters, or type 'r' or 'R' to return menu): ");
       if (fgets (userPhrase, sizeof (userPhrase), stdin) == NULL) {
          printf (YELLOW "Error: Input not received. Exiting function." RESET "\n");
          return;
       }
       userPhrase[strcspn (userPhrase, "\n")] = 0;
-      if (strcmp (userPhrase, "r") == 0 || strcmp (userPhrase, "R") == 0) return;   // Return to menu command
+      if (!strcmp (userPhrase, "r") || !strcmp (userPhrase, "R")) return;
       if (strlen (userPhrase) >= sizeof (userPhrase) - 1) {
          printf (YELLOW "Error: Input exceeds maximum length." RESET "\n");
          while (getchar () != '\n' && getchar () != EOF);
          continue;
       }
-      else printf ("Result: %s\n", CheckPhrasePalindrome (userPhrase) ? GREEN "It's a Palindrome" RESET : RED "Not a Palindrome" RESET);
+      if (strlen (userPhrase) == 0) {
```

Comment:
Always make recommend changes **wherever applicable** **_throughout your program_**.
```suggestion
      if (!strlen (userPhrase)) {
```